### ISSUE TITLE
2D inhomogeneous biharmonic equation

### ISSUE THREAD

**Issue by asdasdqwdqwfsdf** [SciML/NeuralPDE.jl#218]
How to solve a 2D inhomogeneous biharmonic equation with NeuralPDE.jl?

I tried the following code:

```
using NeuralPDE, Flux, ModelingToolkit, GalacticOptim, Optim, DiffEqFlux
using Plots

@parameters x y
@variables u(..)
@derivatives Dxx''~x
@derivatives Dyy''~y
@derivatives Dxxxx''''~x
@derivatives Dyyyy''''~y



# 2D PDE
eq  = Dxxxx(u(x,y)) + Dyyyy(u(x,y)) + 2*Dxx(Dyy(u(x,y)))~ x*y

# Boundary conditions
bcs = [u(0,y) ~ 0.f0, u(1,y) ~ 0.f0,
       u(x,0) ~ 0.f0, u(x,1) ~ 0.f0,
       Dxx(u(0,y)) ~ 0.f0, Dxx(u(1,y)) ~ 0.f0,
       Dyy(u(x,0)) ~ 0.f0, Dyy(u(x,1)) ~ 0.f0
       ]
# Space and time domains
domains = [x ∈ IntervalDomain(0.0,1.0),
           y ∈ IntervalDomain(0.0,1.0)]

# Neural network
dim = 2 # number of dimensions
chain = FastChain(FastDense(dim,16,Flux.σ),FastDense(16,16,Flux.σ),FastDense(16,1))

# Discretization
dx = 0.05
discretization = PhysicsInformedNN(chain,
                                   strategy = GridTraining(dx=dx))

pde_system = PDESystem(eq,bcs,domains,[x,y],[u])
prob = discretize(pde_system,discretization)

#Optimizer
opt = Optim.BFGS()

#Callback function
cb = function (p,l)
    println("Current loss is: $l")
    return false
end

res = GalacticOptim.solve(prob, opt, cb = cb, maxiters=1000)
phi = discretization.phi

xs,ys = [domain.domain.lower:dx/10:domain.domain.upper for domain in domains]


u_predict = reshape([first(phi([x,y],res.minimizer)) for x in xs for y in ys],(length(xs),length(ys)))


plot(xs, ys, u_predict, linetype=:contourf,title = "predict")
```

But it shows such results:

![grafik](https://user-images.githubusercontent.com/58915493/103144417-3dbd9d00-4729-11eb-86f2-e3c4b8772aca.png)

How to derive the solution using NeuralPDE.jl?

**Comment by ChrisRackauckas** [comment]
What hyperparameters were tried? What does the loss trace look like?

**Comment by asdasdqwdqwfsdf** [comment]
> What hyperparameters were tried?

What is your suggestion for the best approximation in this case? 

> What does the loss trace look like?

First loss is: 0.11927169894684062

Last loss is: 2.0872123978938786e-6

It is well known that the CORRECT output DOES **not** ONLY depend on the loss function ...

**Comment by ChrisRackauckas** [comment]
>What is your suggestion for the best approximation in this case?

I haven't looked at this equation yet, so what you tried should get documented to save the time of others in the future. Having a full record of what worked and what didn't work is just good science, and the issues can be thought of as a public lab notebook accumulating information about the repository and its changing behaviors.

>It is well known that the CORRECT output DOES not ONLY depend on the loss function ...

Well duh, but you posted this without any reference to what it was like, so it's pretty hard to judge it without running it. Can you please post the trace? Only showing the first and last doesn't say whether it has converged. It could still be in a downward slope, in which case the answer is to make maxiters higher. Or it can be converged, in which case the answer is something different. The correct output does not only depend on the loss value at the end, so only sharing the loss value at the end is insufficient to diagnose what's going on.

You didn't post any information as to how incorrect the equation is, or how the solving actually went. Can you post the difference from a high fidelity reference solution? That would be good for developing a test case.

**Comment by ChrisRackauckas** [comment]
The issue here was using GridTraining, using any other seems to do well.
